tests/testthat/test-github-connections.R

```diff
@@ -0,0 +1,187 @@
+context("GitHub connections")
+
+## set auth to TRUE if env var GITHUB_PAT is available (eg local testing)
+## set auth to FALSE otherwise (eg on travis, at this point)
+auth <- FALSE
+skip_no_auth <- function() if (!exists("auth") || !auth) skip("no auth")
+
+## why create a temporary test package & repo?
+## if the package lives within tests/testthat like the other test packages,
+## git2r::discover_repository "successfully" discovers the enclosing devtools
+## repository, which prevents usage and testing of uses_git() and use_git()
+## and, therefore, uses_github() and use_github()
+
+create_test_pkg <- function(pkg_name = "testpkg") {
+  capture.output( # necessary because write_dcf() uses cat()
+    suppressMessages({
+      path <- tempfile(pattern="devtools-test-")
+      dir.create(path)
+      path <- file.path(path, pkg_name)
+      create(path, description = list(NULL))
+    })
+  )
+  path
+}
+
+erase_test_pkg <- function(pkg_path) unlink(pkg_path, recursive = TRUE)
+
+github_delete_repo <- function(pkg_path) {
```

Comment:
This still feels risky to me - I'd rather not test adding repos if we have to delete them afterwards.